appinventor/blocklyeditor/src/field_lexical_variable.js

```diff
@@ -410,19 +441,23 @@ Blockly.FieldLexicalVariable.nameNotIn = function(name, nameList) {
       }
     }
   }
-  if (! isConflict) {
-    // There is no conflict; just return name
+
+  if (!isConflict) {
     return name;
-  } else if (! emptySuffixUsed) {
-    // There is a conflict, but empty suffix not used, so use that
+  } else if (!emptySuffixUsed) {
+    // There is a conflict, but empty suffix not used, so use that.
     return namePrefix;
   } else {
     // There is a possible conflict and empty suffix is not an option.
     // First sort the suffixes as numbers from low to high
-    var suffixesAsNumbers = suffixes.map( function (elt, i, arr) { return parseInt(elt,10); } )
+    var suffixesAsNumbers = suffixes.map(function (elt, i, arr) {
+      return parseInt(elt,10);
+    })
     suffixesAsNumbers.sort( function(a,b) { return a-b; } );
-    // Now find smallest number >= 2 that is unused
-    var smallest = 2; // Don't allow 0 or 1 an indices
+
+    // TODO: Is there a different loop method we could use that is more intuitive?
```

Comment:
Yeah I agree it's definitely reasonable... I'm not exactly sure why I felt this way, reading it just felt weird. Like: why is it a while loop instead of a for loop? why is index updated in two places?

I would probably write something closer to this:
```
var smallest = 2;
var length = suffixesAsNumbers.length;
for (var i = 0; i < length; i++) {
  var suffix = suffixesAsNumbers[i];
  if (smallest < suffix) {
    return namePrefix + smallest;
  } else if (smallest == suffix) {
    // This suffix is taken, try the next one.
    smallest++;
  }
}
```

But again, not a priority for this issue! Gotta stay focused.